test/e2e/singlecluster/kuberay_test.go

```diff
@@ -204,6 +272,56 @@ var _ = ginkgo.Describe("Kuberay", func() {
 			}, util.VeryLongTimeout, util.Interval).Should(gomega.Succeed())
 		})
 
+		ginkgo.By("Waiting for 3 pods in rayjob namespace", func() {
+			// 3 rayjob pods: head, worker, submitter job
+			gomega.Eventually(func(g gomega.Gomega) {
+				podList := &corev1.PodList{}
+				g.Expect(k8sClient.List(ctx, podList, client.InNamespace(ns.Name))).To(gomega.Succeed())
+				g.Expect(podList.Items).To(gomega.HaveLen(3), "Expected exactly 3 pods in rayjob namespace")
+				// Count pods that have "workers" in their name
+				workerPodCount := countRunningWorkerPods(podList)
+				g.Expect(workerPodCount).To(gomega.Equal(1), "Expected exactly 1 pod with 'workers' in the name")
+			}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("Waiting for 2 workloads", func() {
+			// 2 workloads: one for the ray cluster, another for the submitter job created by the ray job
+			gomega.Eventually(func(g gomega.Gomega) {
+				workloadList := &kueue.WorkloadList{}
+				g.Expect(k8sClient.List(ctx, workloadList, client.InNamespace(ns.Name))).To(gomega.Succeed())
+				g.Expect(workloadList.Items).To(gomega.HaveLen(2), "Expected exactly 2 workloads")
+			}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("Waiting for 5 workers due to scaling up", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				podList := &corev1.PodList{}
+				g.Expect(k8sClient.List(ctx, podList, client.InNamespace(ns.Name))).To(gomega.Succeed())
+				// Count pods that have "workers" in their name
+				workerPodCount := countRunningWorkerPods(podList)
+				g.Expect(workerPodCount).To(gomega.Equal(5), "Expected exactly 5 pods with 'workers' in the name")
```

Comment:
Actually, it is better if this helper returns Pod names rather than just count.

Then I think we should verify the new 5 pods is a superset of the old Pods. Otherwise maybe they were all re-created. Similarly, on scale down we would like to check that the new set of pods is a subset of previous Pods.

Let me know what you think, but I expect this will make the tests proving the scaling really works. Otherwise it was possible also in the past with suspending / re admitting the Job.

Reply:
Update PR to check pod names to be superset after scaling up/down.